core/src/main/scala/org/apache/spark/scheduler/ResultTask.scala

```diff
@@ -65,4 +65,6 @@ private[spark] class ResultTask[T, U](
   override def preferredLocations: Seq[TaskLocation] = preferredLocs
 
   override def toString = "ResultTask(" + stageId + ", " + partitionId + ")"
+
+  override def canEqual(other: Any): Boolean = other.isInstanceOf[ResultTask[T, U]]
```

Comment:
yean, I know that.  and in that class, it has no need to add parameter on class level, it could only use in function level `run` or `runContext`.